### DISCUSSION TITLE
how to use TextAlignment with command line

### DISCUSSION THREAD

**Discussion by viratsoft** [ImageMagick/ImageMagick#4467] [Help]
hi,
 i want to auto center text according width of text with angle.

i want to write auto text on image that auto center with specific x and y point (Ex. 450 and 500) with angle and text automatically like bellow image.
![Untitled-ok](https://user-images.githubusercontent.com/49095830/141240592-344c9c1f-cb2b-4146-b4fd-5cf853a33a33.png)

 in php `$draw->setTextAlignment(Imagick::ALIGN_CENTER);` work fine.
currently i use like bellow but that  start from 450x500 point.
`convert -annotate 320x320+450+500 "Sample Text"`

how to use setTextAlignment in command line?

**Comment by snibgo** [comment]
I haven't checked this, but I suppose PHP setTextAlignment is equivalent to "text-align" in a "-draw" string, such as:
```
magick -size 100x100 xc:#88f -draw "text-align center text 50,50 M12345" x.png
```

**Reply by fmw42** [reply]
Interesting. But I do not see the text-align option at https://legacy.imagemagick.org/script/magick-vector-graphics.php. Where did you find that? Is it documented?

**Comment by snibgo** [comment]
I read it somewhere, but I forget where. It may have been the source code draw.c function RenderMVGContent().

**Comment by fmw42** [comment]
**Developers -- Please document text-align option for -draw in https://legacy.imagemagick.org/script/magick-vector-graphics.php and https://imagemagick.org/script/command-line-options.php**
